pyrokinetics/gk_code/GKInputGS2.py

```diff
@@ -212,6 +212,99 @@ def get_local_species(self):
         local_species.normalise()
         return local_species
 
+    def _read_single_grid(self):
+        return {
+            "nky": 1,
+            "nkx": 1,
+            "ky": self.data["kt_grids_single_parameters"]["aky"] / sqrt2,
+            "kx": 0.0,
+            "theta0": self.data["kt_grids_single_parameters"].get("theta0", 0.0),
+        }
+
+    def _read_range_grid(self):
+        range_options = self.data["kt_grids_range_parameters"]
+        nky = range_options.get("naky", 1)
+
+        ky_min = range_options.get("aky_min", "0.0")
+        ky_max = range_options.get("aky_max", "0.0")
+
+        spacing_option = range_options.get("kyspacing_option", "linear")
+        if spacing_option == "default":
+            spacing_option = "linear"
+
+        ky_space = np.linspace if spacing_option == "linear" else np.logspace
+
+        ky = ky_space(ky_min, ky_max, nky) / sqrt2
+
+        return {
+            "nky": nky,
+            "nkx": 1,
```

Comment:
This could be >1

Reply:
Good point, I left this for the time being, because I haven't quite got my head round how the `kx` grid is set. It looks like there's at least four different ways

Reply:
When we have a box with multiple `kx` and `ky` we should use the method in the nonlinear case. This is specifically for single mode analysis. Might need better logic for this though

Reply:
Sorry, just to be clear and so we're all on the same page, I've rearranged things a little to be able to handle three different GS2 grids:

- `single`, which was what the linear case previously used
- `range`, which is new (to pyrokinetics) and another option for linear runs, and could have multiple `kx` modes
- `box`, which is what the nonlinear case previously used

Reply:
Is `nkx` basically set by `gs2['kt_grids_range_parameters']['ntheta0']`